packages/sdk-js/src/GrowthBook.ts

```diff
@@ -1006,6 +1006,12 @@ export class GrowthBook<
       document.head.appendChild(s);
       undo.push(() => s.remove());
     }
+    if (changes.js) {
+      const script = document.createElement("script");
+      script.innerHTML = changes.js;
+      document.body.appendChild(script);
+      undo.push(() => script.remove());
+    }
```

Comment:
This could potentially run the script multiple times, right?  Is that something we need to worry about?  One example use case for js injection is adding a chat widget like Intercom to the page, which this could break.

One potential solution is to get rid of the "undo" part and instead keep track of which scripts we've already added so we don't add it more than once to a page.

Reply:
This is a great point. I thought about this but have two questions:

- Do we undo and redo changes inside a page after it's loaded? Based on your suggestion, it seems like yes. (If so, does a mutation observer drive this?)
- Do we undo changes when we navigate to a page that is NOT targeted by the visual experiment's URL pattern?

In the 2nd case it seems to me that having an undo for the JS changes is necessary, so I wonder if we need to support both cases.

Reply:
Per our convo - Added the ability to force re-renders and leverage through `setURL`  call